lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -31,8 +32,33 @@
 from ansible.module_utils.six import string_types
 from ansible.utils.version import SemanticVersion
 
+from collections.abc import Set
 from resolvelib import AbstractProvider
 
+PINNED_CANDIDATE_ATTRS = ('fqcn', 'ver', 'src', 'type')
+
+
+class PinnedCandidateRequests(Set):
+    def __init__(self, candidates):
+        self.candidates = set(candidates)
```

Comment:
This seems to be internal, so I'd add a leading underscore.

Reply:
Fixed.